src/platform/platform_worker.c

```diff
@@ -556,16 +556,18 @@ CxPlatProcessEvents(
     )
 {
     CXPLAT_CQE Cqes[16];
+    CXPLAT_CQE* CurrentCqe = Cqes;
     uint32_t CqeCount = CxPlatEventQDequeue(&Worker->EventQ, Cqes, ARRAYSIZE(Cqes), State->WaitTime);
+    uint32_t CurrentCqeCount = CqeCount;
     InterlockedFetchAndSetBoolean(&Worker->Running);
     if (CqeCount != 0) {
 #if DEBUG // Debug statistics
         Worker->CqeCount += CqeCount;
 #endif
         State->NoWorkCount = 0;
-        for (uint32_t i = 0; i < CqeCount; ++i) {
-            CXPLAT_SQE* Sqe = CxPlatCqeGetSqe(&Cqes[i]);
-            Sqe->Completion(&Cqes[i]);
+        while (CqeCount > 0) {
+            CXPLAT_SQE* Sqe = CxPlatCqeGetSqe(CurrentCqe);
+            Sqe->Completion(&CurrentCqe, &CurrentCqeCount);
```

Comment:
Every SQE has its own completion function. Why are you changing that?

Reply:
The new completion function supports batching completions, which is better than calling Foo() 100 times, and having Foo() do boilerplate (lock, submit, etc.) each time.

Reply:
Batching of what exactly though? If you have a list of 10 completion functions to call, what is being batched? The callbacks are all outside of a lock, right?

Reply:
The `CurrentCqe` and `CurrentCqeCount` are both `_Inout_` parameters, so the routine can consume [1, CurrentCqeCount] CQEs.

Here's the only routine taking advantage:
https://github.com/microsoft/msquic/blob/0752c6da0e1835622ba7e1a46d7d652de111194f/src/platform/datapath_iouring.c#L2106-L2116